lib/trento/domain/cluster/cluster.ex

```diff
@@ -136,20 +136,35 @@ defmodule Trento.Domain.Cluster do
   def execute(
         %Cluster{
           cluster_id: cluster_id
-        },
+        } = cluster,
         %SelectChecks{
           checks: selected_checks
         }
       ) do
-    [
-      %ChecksSelected{
-        cluster_id: cluster_id,
-        checks: selected_checks
-      }
-    ]
+    cluster
+    |> Multi.new()
+    |> Multi.execute(fn _ ->
+      [
+        %ChecksSelected{
+          cluster_id: cluster_id,
+          checks: selected_checks
+        }
+      ]
+    end)
+    |> Multi.execute(fn cluster -> maybe_emit_cluster_health_changed_event(cluster) end)
   end
 
   # Request checks execution
+  def execute(
+        %Cluster{
+          cluster_id: cluster_id,
+          selected_checks: selected_checks
+        },
+        %RequestChecksExecution{cluster_id: cluster_id}
+      )
+      when selected_checks == [],
```

Comment:
the guard `when` is not needed, since you can pattern match with
```elixir
  %Cluster{
          cluster_id: cluster_id,
          selected_checks: []
        }
```